src/docker/shared.rs

```diff
@@ -499,8 +512,49 @@ pub(crate) fn docker_envvars(
     Ok(())
 }
 
-pub(crate) fn docker_cwd(docker: &mut Command, paths: &DockerPaths) -> Result<()> {
+fn mount_to_ignore_cargo_config(
+    docker: &mut Command,
+    paths: &DockerPaths,
+    ignore_cargo_config: bool,
+) -> Result<()> {
+    let check_mount =
+        |cmd: &mut Command, host: &Path, mount: &Path, relpath: &Path| -> Result<()> {
+            let cargo_dir = relpath.join(".cargo");
+            if host.join(&cargo_dir).exists() {
+                // this is fine, since it has to be a POSIX path on the mount.
+                cmd.args(&["-v", &mount.join(&cargo_dir).as_posix()?]);
+            }
+
+            Ok(())
+        };
+    let is_cargo_home_parent = paths
+        .cargo_home()
+        .parent()
+        .map(|x| x == paths.host_root())
+        .unwrap_or_default();
```

Comment:
This check is probably unnecessary. This check should probably be removed. Thoughts? This won't effect mounting the cargo home directory to `/cargo`.

Reply:
I've removed this check, since I've tested to see if invoking a build from the parent of `CARGO_HOME` works, and it does, and doesn't touch `CARGO_HOME` at all. It just prevents that directory from being mounted at the same path as on the host, but still mounts it with all the data at `/cargo`.